query_test.go

```diff
@@ -77,51 +77,59 @@ func TestGormLike_Initialize_TriggersLikingCorrectly(t *testing.T) {
+				"id": "%22%",
+			},
+			query:    defaultQuery,
+			existing: []ObjectA{{ID: uuid.MustParse("573fc37d-622c-4d41-b48b-201bf195a1c9"), Name: "jessica", Age: 53}, {ID: uuid.MustParse("66df272f-5667-4472-950a-2269833fdbab"), Name: "amy", Age: 20}, {ID: uuid.MustParse("3e3b8dbc-37a3-4113-b05c-6c1d5a3ad5fa"), Name: "John", Age: 25}},
+			expected: []ObjectA{{ID: uuid.MustParse("573fc37d-622c-4d41-b48b-201bf195a1c9"), Name: "jessica", Age: 53}, {ID: uuid.MustParse("66df272f-5667-4472-950a-2269833fdbab"), Name: "amy", Age: 20}},
 		},
 		"more complex like query": {
 			filter: map[string]any{
-				"name": "%a%",
+				"name": "a%",
 				"age":  20,
 			},
 			query:    defaultQuery,
-			existing: []ObjectA{{Name: "jessica", Age: 53}, {Name: "amy", Age: 20}, {Name: "John", Age: 25}},
-			expected: []ObjectA{{Name: "amy", Age: 20}},
+			existing: []ObjectA{{ID: uuid.MustParse("25c9b291-951c-4213-b71e-6e3c63ecbe72"), Name: "jessica", Age: 53}, {ID: uuid.MustParse("943ce561-2ae7-40b8-8540-dc49ca84e400"), Name: "amy", Age: 20}, {ID: uuid.MustParse("2d6d5829-cb2b-4a8a-9d09-70ea46b9a62e"), Name: "John", Age: 25}},
+			expected: []ObjectA{{ID: uuid.MustParse("943ce561-2ae7-40b8-8540-dc49ca84e400"), Name: "amy", Age: 20}},
 		},
 		"multi-value, all like queries": {
 			filter: map[string]any{
 				"name": []string{"%a%", "%o%"},
 			},
 			query:    defaultQuery,
-			existing: []ObjectA{{Name: "jessica", Age: 53}, {Name: "amy", Age: 20}, {Name: "John", Age: 25}},
-			expected: []ObjectA{{Name: "jessica", Age: 53}, {Name: "amy", Age: 20}, {Name: "John", Age: 25}},
+			existing: []ObjectA{{ID: uuid.MustParse("225c43b8-4645-4603-a362-721f4af370ab"), Name: "jessica", Age: 53}, {ID: uuid.MustParse("b680ef6f-c894-4a4f-a90d-245fb3284c75"), Name: "amy", Age: 20}, {ID: uuid.MustParse("f76c692d-4991-4da2-880e-38e19ca3261e"), Name: "John", Age: 25}},
+			expected: []ObjectA{{ID: uuid.MustParse("225c43b8-4645-4603-a362-721f4af370ab"), Name: "jessica", Age: 53}, {ID: uuid.MustParse("b680ef6f-c894-4a4f-a90d-245fb3284c75"), Name: "amy", Age: 20}, {ID: uuid.MustParse("f76c692d-4991-4da2-880e-38e19ca3261e"), Name: "John", Age: 25}},
 		},
 		"more complex multi-value, all like queries": {
 			filter: map[string]any{
 				"name":  []string{"%a%", "%o%"},
 				"other": []string{"%ooo", "aaa%"},
 			},
 			query:    defaultQuery,
-			existing: []ObjectA{{Name: "jessica", Age: 53, Other: "aaaooo"}, {Name: "amy", Age: 20, Other: "aaaooo"}, {Name: "John", Age: 25, Other: "aaaooo"}},
-			expected: []ObjectA{{Name: "jessica", Age: 53, Other: "aaaooo"}, {Name: "amy", Age: 20, Other: "aaaooo"}, {Name: "John", Age: 25, Other: "aaaooo"}},
+			existing: []ObjectA{{ID: uuid.MustParse("77c5e4f2-30a9-459b-8178-0a3b61369cdc"), Name: "jessica", Age: 53, Other: "aaaooo"}, {ID: uuid.MustParse("909ec358-ee69-4f74-8e70-069311125e8a"), Name: "amy", Age: 20, Other: "aaaooo"}, {ID: uuid.MustParse("37ac3821-c4e7-4004-b2ed-2cdc7c9ffb53"), Name: "John", Age: 25, Other: "aaaooo"}},
```

Comment:
Could you multiline this data? I don't think the uuid makes it more readable

Reply:
Or perhaps the IDs could be removed as to not pollute the test data where it isn't needed

Reply:
made the test inputs more readable